### ISSUE TITLE
Can I use with torch.no_grad() in the forward of dropblock.py?

### ISSUE THREAD

**Issue by StrongerzZ** [miguelvr/dropblock#41]
As the title.
Can someone help me answer it?
thanks!

**Comment by miguelvr** [comment]
During validation / testing, yes you should. 

Can you elaborate on the issue?

**Comment by StrongerzZ** [comment]
Thanks for the answer, I also want to ask why can't use with torch.no_grad() during training, because I feel it does not require gradient tracking.
Maybe i'm wrong...

**Comment by miguelvr** [comment]
You can't use torch.no_grad() in a subset of your graph because it will break it. Either you use it in your whole model, or you don't. The only way that I envision that being possible is if dropblock is applied to your inputs, before the rest of the model. 

In any case this is not a dropblock issue, it's a pytorch issue.

**Comment by StrongerzZ** [comment]
I understand, thank you very much！ @miguelvr
